cJSON.c

```diff
@@ -452,20 +431,22 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
     {
         length = sprintf((char*)number_buffer, "null");
     }
-    else if ((fabs(floor(d) - d) <= DBL_EPSILON) && (fabs(d) < 1.0e60))
+    /* This checks for negative zero */
+    else if (d == 0)
     {
-        /* integer */
-        length = sprintf((char*)number_buffer, "%.0f", d);
-        trim_zeroes = false; /* don't remove zeroes for "big integers" */
-    }
-    else if ((fabs(d) < 1.0e-6) || (fabs(d) > 1.0e9))
-    {
-        length = sprintf((char*)number_buffer, "%e", d);
-        trim_zeroes = false; /* don't remove zeroes in engineering notation */
+        length = sprintf((char*)number_buffer, "0");
```

Comment:
Why print `-0` as `0`? RFC7159 allows negative zero and it can be represented by doubles internally, so we can as well print it as is.